contracts/AnchoredView/AnchoredView.sol

```diff
@@ -110,29 +110,29 @@ contract AnchoredView is SymbolConfiguration {
         }
 
         // load usdc for using in loop to convert anchor prices to dollars
-        uint256 usdcPrice = readAnchor(cUsdcAddress);
+        uint usdcPrice = readAnchor(cUsdcAddress);
 
         // Try to update the view storage
         for (uint i = 0; i < symbols.length; i++) {
             string memory symbol = symbols[i];
+
             address tokenAddress = getCTokenAddress(symbol);
-            uint256 sourcePrice = priceData.getPrice(source, symbol);
-            uint256 anchorPrice = getAnchorInUsd(tokenAddress, usdcPrice);
 
-            if (anchorPrice == 0 || tokenAddress == cUsdcAddress || tokenAddress == cUsdtAddress) {
+            uint sourcePrice = priceData.getPrice(source, symbol);
+            uint anchorPrice = getAnchorInUsd(tokenAddress, usdcPrice);
+
+            if (tokenAddress == cUsdcAddress || tokenAddress == cUsdtAddress || anchorPrice == 0)  {
                 emit PriceGuarded(symbol, sourcePrice, anchorPrice);
             } else {
-                uint256 anchorRatioMantissa = mul(sourcePrice, 100e16) / anchorPrice;
-                // Only update the view's price if the source price is within a bound
-                if (anchorRatioMantissa <= upperBoundAnchorRatio && anchorRatioMantissa >= lowerBoundAnchorRatio) {
+                uint anchorRatio = mul(sourcePrice, 100e16) / anchorPrice;
+                bool withinAnchor = anchorRatio <= upperBoundAnchorRatio && anchorRatio >= lowerBoundAnchorRatio;
+
+                if (withinAnchor || !anchored) {
```

Comment:
Maybe `!anchored` check and price update can be moved even above `getAnchorInUsd(tokenAddress, usdcPrice);` ? 
Otherwise all mantissa checks are done for nothing if it's !anchored

Reply:
to question 1, at first !anchored was it's own block after this one since I was trying to skip that storage read if it was withinAnchor, that should probably go back to how it was.

question 2, also a good point..

Reply:
actually , point 1, this boolean short circuit will skip the storage read on !anchored in most cases, so I think the logic is best there.

refactoring to move the 0 check into the within anchor check so we still get the benefit of the cut anchor